src/utils/keyRotation.ts

```diff
@@ -0,0 +1,30 @@
+import { Bindings } from '../bindings';
+import cronParser from 'cron-parser';
+
+export function shouldRotateKey(date: Date, env: Bindings): boolean {
+	const utcDate = new Date(date.toISOString());
+
+	if (env.ROTATION_CRON_STRING) {
+		const result = matchCronTime(env.ROTATION_CRON_STRING, utcDate);
+		return result;
+	}
+	return false;
+}
+
+function matchCronTime(cronString: string, date: Date): boolean {
+	const options = {
+		currentDate: date.toISOString(),
+		tz: 'UTC',
+	};
+
+	const interval = cronParser.parseExpression(cronString, options);
+	const prevDate = interval.prev().toDate();
+	const nextDate = interval.next().toDate();
+
+	// Round the current date to the nearest second
+	const roundedDate = new Date(Math.round(date.getTime() / 1000) * 1000);
```

Comment:
I think this is equivalent
```suggestion
	// Round the current date to the nearest second
	date.setUTCMilliseconds(1000);
```